### DISCUSSION TITLE
Provide separate `export_app` to re-export previous `build_app`.

### DISCUSSION THREAD

**Discussion by lhunath** [fastlane/fastlane#19220] [Ideas]
When `build_app` builds an iOS app, it performs two steps:
1. `xcodebuild clean archive`
2. `xcodebuild -exportArchive`

The result is an IPA that can be used by the lane's future steps.

For instance, it could also be uploaded to the device for on-device verification.
The build could also be uploaded to Apple for TestFlight or submission as a release.

The trouble is that for uploading builds to TestFlight or for App Store submission, they should be signed with an App Store provisioning profile and certificate, while for use in other actions such as on-device testing, they should be signed as developer or ad-hoc.

`xcodebuild -exportArchive` should be able to handle this fine as it can re-sign archives, but currently `build_app` wraps both of these steps in one, making it impossible to have a lane that does both local on-device verification of the build product & uploading of the verified build product to App Store Connect.

A way to address this might be to provide an optional `export_app` which performs only the export step using a previously built archive.  That way the first `build_app` could export the app for local use and a second `export_app` could make an IPA available for Apple/public use.

**Comment by lucgrabowski** [comment]
Hi, I think that `skip_build_archive` (Export ipa from previously built xcarchive. Uses archive_path as source) might be the option that you're looking for. 
https://docs.fastlane.tools/actions/gym/
